java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -24,18 +24,91 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 import java.util.function.Consumer;
 
+import static ai.rapids.cudf.HostColumnVector.OFFSET_SIZE;
+
 /**
  * This class represents the immutable vector of data.  This class holds
  * references to device(GPU) memory and is reference counted to know when to release it.  Call
  * close to decrement the reference count when you are done with the column, and call incRefCount
  * to increment the reference count.
  */
-public final class ColumnVector implements AutoCloseable, BinaryOperable {
+public final class ColumnVector implements AutoCloseable, BinaryOperable, ColumnViewPointerAccess {
   private static final Logger log = LoggerFactory.getLogger(ColumnVector.class);
 
+  public class DeviceColumnViewPointerAccess implements ColumnViewPointerAccess<BaseDeviceMemoryBuffer> {
+
+    protected long viewHandle;
+
+    public DeviceColumnViewPointerAccess(long viewHandle) {
+      this.viewHandle = viewHandle;
```

Comment:
This is a memory leak. In the C++ code the only way to get a view out is to copy it.  Which the native code does, but we don't have a close method on this that will free the host memory associated with the column view that we just copied out. I was hopeful that we could just reuse the same pointer that is stored in the `_children` vector, but that probably violates something with the abstraction for the C++ layer, so we are going to need to be sure that this gets closed each time it is used.

Reply:
I have made some changes that should close the column view. Let me know what you think.

Reply:
@revans2 Please see the use of .close() on the ColumnViewPointerAccess as that is how I am tackling it right now. Please correct  me if I missed something.

Reply:
I would really prefer to have us use a try/finally where ever possible.  But this works.